lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -552,54 +561,86 @@
+        // Use matching groups only when needed
         const LANG_TAG_BASE = `\\b(${language})\\b`         +                   // langtag       = language
                               `\\b(?:-(${script}))?\\b`     +                   //                 ["-" script]
                               `\\b(?:-(${region}))?\\b`     ;                   //                 ["-" region]
-        const LANG_TAG_EXT  = `\\b(?:-${variant})*\\b`      +                   //                 *("-" variant)
+        const LANG_TAG_EXT  = `\\b((?:-${variant})*)\\b`    +                   //                 *("-" variant)
                               `\\b((?:-${extension})*)\\b`  +                   //                 *("-" extension)
                               `\\b(?:-${privateuse})?\\b`   ;                   //                 ["-" privateuse]
         const langtag       = `\\b${LANG_TAG_BASE}\\b${LANG_TAG_EXT}\\b`;
 
         const LANG_TAG      = `\\b(?:${langtag}|${privateuse}|${grandfathered})\\b`;  // Language-Tag  = ...
 
-        LANG_TAG_BASE_RE    = new RegExp(LANG_TAG_BASE, 'i'); // [1] language; [2] script; [3] region
-        LANG_TAG_EXT_RE     = new RegExp(LANG_TAG_EXT,  'i'); //                                       [1] extensions /((${extension})-)*/
-        LANG_TAG_RE         = new RegExp(LANG_TAG,      'i'); // [1] language; [2] script; [3] region; [4] extensions
-    })();
+        // Use ^ and $ to enforce that the entire input string is a langtag
+        const LANG_TAG_BASE_RE    = new RegExp(`^${LANG_TAG_BASE}$`, 'i'); // [1] language; [2] script; [3] region;
+        const LANG_TAG_EXT_RE     = new RegExp(`^${LANG_TAG_EXT}$`,  'i'); //                                       [1] variants; [2] extensions;
+        const LANG_TAG_RE         = new RegExp(`^${LANG_TAG}$`,      'i'); // [1] language; [2] script; [3] region; [4] variants; [5] extensions;
 
-    /**
-     * Returns an object representing the language, script, region, extension, and base of a language tag,
-     * or null if the language tag isn't valid.
-     *
-     * @param {String} langtag a candidate BCP47 langtag
-     */
-    const parseLangtag = function (langtag) {
-        const parts = _.match(langtag, LANG_TAG_RE);
-        if (!parts || !parts[1]) {
-            return null;
-        }
+        const parsedLangtagCache = new IntlCache();
+        return function (langtag) {
+            const cached = parsedLangtagCache.get(langtag);
+            if (cached) {
+                return cached;
+            }
 
-        const ret = { language: parts[1], base: parts[1] };
-        if (parts[2]) {
-            ret.script = parts[2];
-            ret.base += "-" + parts[2];
-        }
+            const parts = _.match(langtag, LANG_TAG_RE);
+            if (!parts) {
+                return null;
+            }
 
-        if (parts[3]) {
-            ret.region = parts[3];
-            ret.base += "-" + parts[3];
-        }
+            const ret = _.create();
```

Comment:
>create [](start = 26, length = 6)

now that I have some distance from it, _.create() is not that clear about what it is NOR how it related to the spec. maybe record is a better name?

Reply:
The name is short only because thats the convention I have been using -- the `_` basically stands in for String, Object, etc. I can change it to objectCreate() (or createObject, but that puts us even further from convention)